python/src/sphericart/spherical_harmonics.py

```diff
@@ -84,23 +73,75 @@ def compute(
 
         sph = np.empty((n_samples, (self._l_max + 1) ** 2), dtype=xyz.dtype)
         sph_length = n_samples * (self._l_max + 1) ** 2
-        if gradients:
-            dsph = np.empty((n_samples, 3, (self._l_max + 1) ** 2), dtype=xyz.dtype)
-            dsph_length = n_samples * 3 * (self._l_max + 1) ** 2
-        else:
-            dsph = None
-            dsph_length = 0
 
         if xyz.dtype == np.float64:
             xyz_ptr = xyz.ctypes.data_as(ctypes.POINTER(ctypes.c_double))
             sph_ptr = sph.ctypes.data_as(ctypes.POINTER(ctypes.c_double))
 
-            if gradients:
-                dsph_ptr = dsph.ctypes.data_as(ctypes.POINTER(ctypes.c_double))
-            else:
-                dsph_ptr = None
-
             self._lib.sphericart_compute_array(
+                self._calculator, xyz_ptr, xyz_length, sph_ptr, sph_length
+            )
+        elif xyz.dtype == np.float32:
+            xyz_ptr = xyz.ctypes.data_as(ctypes.POINTER(ctypes.c_float))
+            sph_ptr = sph.ctypes.data_as(ctypes.POINTER(ctypes.c_float))
+
+            self._lib.sphericart_compute_array_f(
+                self._calculator_f, xyz_ptr, xyz_length, sph_ptr, sph_length
+            )
+
+        return sph
+
+    def compute_with_gradients(self, xyz: np.ndarray) -> Tuple[np.ndarray]:
```

Comment:
```suggestion
    def compute_with_gradients(self, xyz: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
```

the syntax for typing tuples is strange: Tuple[A] is a single element tuple, Tuple[A, B, A] is a three elements tuple, Tuple[A, ...] is a tuple of unknown length containing only A